cas-client-integration-jetty/src/main/java/org/jasig/cas/client/jetty/CasAuthentication.java

```diff
@@ -0,0 +1,45 @@
+package org.jasig.cas.client.jetty;
+
+import org.eclipse.jetty.security.UserAuthentication;
+import org.jasig.cas.client.validation.Assertion;
+
+/**
+ * CAS-specific user authentication.
+ *
+ * @author Marvin S. Addison
+ */
+public class CasAuthentication extends UserAuthentication {
+
+    /** CAS authenticator that produced this authentication. */
+    private final CasAuthenticator authenticator;
+
+    /** CAS ticket that was successfully validated to permit authentication. */
+    private final String ticket;
+
+
+    /**
+     * Creates a new instance.
+     *
+     * @param authenticator The authenticator that produced this authentication.
+     * @param ticket The CAS ticket that was successfully validated to permit authentication.
+     * @param assertion The CAS assertion produced from successful ticket validation.
+     */
+    public CasAuthentication(final CasAuthenticator authenticator, final String ticket, final Assertion assertion) {
+        super(authenticator.getAuthMethod(), new CasUserIdentity(assertion, authenticator.getRoleAttribute()));
+        assert ticket != null : "Ticket cannot be null";
```

Comment:
For consistency, sure, but I think assert is better in every way. We might consider a separate issue to move to the keyword across the board.